### DISCUSSION TITLE
Database startup & recreation

### DISCUSSION THREAD

**Discussion by computinglife** [rqlite/rqlite#1822] [General]
Hi, 

The docs say this "SQLite database is completely recreated everytime rqlited starts, using the information stored in the Raft log" 

I found this interesting - Can you please shed some light on why this is done and how long this might take, say on a database that has a GB or more worth of data ? 

Wouldnt this require logs going all the way back to the time when DB was created not to mention the extra storage required? 

I would imagine that way its easier to keep only raft logs upto the last synced point in the cluster - that way only some amount of data on the disk needs to be maintained ? 

Can you please explain why this is done?

Thanks
Arun

**Comment by otoolep** [comment]
By “Raft log” we actually mean “Raft log and any Snapshots”. We only keep the logs after the last “snapshot” point, as you say. 
The best way to answer this question is to study the Raft paper (or any presentation on Raft) and rqlite design doc resources. 
https://rqlite.io/docs/design/ 


Sent from Yahoo Mail for iPhone


On Wednesday, June 26, 2024, 10:15 AM, computinglife ***@***.***> wrote:




Hi,

The docs say this "SQLite database is completely recreated everytime rqlited starts, using the information stored in the Raft log"

I found this interesting - Can you please shed some light on why this is done and how long this might take, say on a database that has a GB or more worth of data ?

Wouldnt this require logs going all the way back to the time when DB was created not to mention the extra storage required?

I would imagine that way it’s easier to keep only raft logs upto the last synced point in the cluster - that way only some amount of data on the disk needs to be maintained ?




Can you please explain why this is done?

Thanks
Arun

**Comment by computinglife** [comment]
Ah ok. Thank you. Thats clear now.
